client/src/components/EmojiPicker.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+import React from 'react';
+import ReactEmoji from "react-emoji";
+import { filterEmojis } from "../utils/emojis";
+import 'styles/EmojiPicker.scss';
+
+const emojify = (emojis, highlightedIndex) => {
+    const emojiOpts = {
+      emojiType: 'emojione'
+    };
+    let counter = 0;
+    return emojis.map(emoji => {
+            let elementClass = '';
+            const element = emoji.element;// ReactEmoji.emojify(emoji, emojiOpts)[0];
+            if (counter === highlightedIndex)
+                elementClass = 'selected';
+            return (<li key={counter++}
+                        className={elementClass}>
+                        {element}
+                    </li>);
+        }
+    );
+};
+
```

Comment:
We can now simplify the `emojify`function. It can take in the list of supported emoji (as generated at the beginning) and outputs the emojis as React elements:
```javascript
const emojify = (emojis, highlightedIndex) => {
  const emojiOpts = { emojiType: 'emojione' };
  return emojis
    .map((e) => ReactEmoji.emojify(e.shortname, emojiOpts))
    .map((e, index) => <li key={index} className={index === highlightedIndex ? 'selected' : ''}>{e}</li>);
};
````

Notice how we can replace the counter with `index` number from the map function. Very handy for these kind of cases :)